pkg/scheduler/util/utils_test.go

```diff
@@ -226,20 +236,106 @@ func TestPatchPodStatus(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "retry patch request when an 'connection refused' error is returned",
+			client: func() *clientsetfake.Clientset {
+				client := clientsetfake.NewSimpleClientset()
+
+				reqcount := 0
+				client.PrependReactor("patch", "pods", func(action clienttesting.Action) (bool, runtime.Object, error) {
+					defer func() { reqcount++ }()
+					if reqcount == 0 {
+						// return an connection refused error for the first patch request.
+						return true, &v1.Pod{}, fmt.Errorf("connection refused: %w", syscall.ECONNREFUSED)
+					}
+					if reqcount == 1 {
+						// not return error for the second patch request.
+						return false, &v1.Pod{}, nil
+					}
+
+					// return error if requests comes in more than three times.
+					return true, nil, errors.New("requests comes in more than three times.")
+				})
+
+				return client
+			}(),
+			pod: v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "ns",
+					Name:      "pod1",
+				},
+				Spec: v1.PodSpec{
+					ImagePullSecrets: []v1.LocalObjectReference{{Name: "foo"}},
+				},
+			},
+			statusToUpdate: v1.PodStatus{
+				Conditions: []v1.PodCondition{
+					{
+						Type:   v1.PodScheduled,
+						Status: v1.ConditionFalse,
+					},
+				},
+			},
+		},
+		{
+			name: "only 4 retries at most",
+			client: func() *clientsetfake.Clientset {
+				client := clientsetfake.NewSimpleClientset()
+
+				reqcount := 0
+				client.PrependReactor("patch", "pods", func(action clienttesting.Action) (bool, runtime.Object, error) {
+					defer func() { reqcount++ }()
+					if reqcount >= 4 {
+						// return error if requests comes in more than six times.
```

Comment:
four?

But why change the error? It should stop retrying even if the error is the same.

Reply:
Thanks. Fixed typo. 🙏 

---

> But why change the error? It should stop retrying even if the error is the same.

This "requests comes in more than four times." error shouldn't be returned in the expected scenario.

The expected scenario is:
1. PatchPodStatus func sends a patch request to fake client.
2. fake client returns "connection refused" error
3. PatchPodStatus func retry (1) three times and fake client returns "connection refused" errors for these three requests.
4. PatchPodStatus func gives up retrying and returns "connection refused" error. (the test will check if the returned error is "connection refused" or not.)

`PrependReactor` returns "connection refused" error only four times and if there is a bug like "the system retries forever", the "requests comes in more than four times." error will be returned and this test case will fail. (because the test will check if the returned error is "connection refused" or not.)